staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -179,8 +217,9 @@ func ExtractList(obj runtime.Object) ([]runtime.Object, error) {
 	list := make([]runtime.Object, items.Len())
 	for i := range list {
 		raw := items.Index(i)
-		switch item := raw.Interface().(type) {
-		case runtime.RawExtension:
+		switch {
+		case raw.Type() == rawExtensionObjectType:
```

Comment:
the type() accessor and these checks can be calculated once as an optimization ... I did that as a commit in https://github.com/liggitt/kubernetes/commits/extract_listobject_squash and it resolved a slight CPU regression

Reply:
Greate optimization, each element type in objList is the same. There is no need to perform type judgment every time in the for-loop.